src/coreclr/src/jit/morph.cpp

```diff
@@ -8159,7 +8159,8 @@ void Compiler::fgMorphRecursiveFastTailCallIntoLoop(BasicBlock* block, GenTreeCa
                 var_types lclType            = varDsc->TypeGet();
                 bool      isUserLocal        = (varNum < info.compLocalsCount);
                 bool      structWithGCFields = ((lclType == TYP_STRUCT) && varDsc->GetLayout()->HasGCPtr());
-                if (isUserLocal || structWithGCFields)
+                bool      hadSuppressedInit  = varDsc->lvSuppressedZeroInit;
+                if (isUserLocal || structWithGCFields || hadSuppressedInit)
```

Comment:
I think this is too conservative. Consider this case:
`info.compInitMem` is false
`compSuppressedZeroInit` is true
`hadSuppressedInit` is false

In this case we shouldn't add zero init regardless of the value of `isUserLocal` and `structWithGCFields`.

Reply:
Sure, will update.